### PR TITLE
Mask generation for training dataset

### PR DESCRIPTION
**Pull request by navjot-nangia**
*** working on it ***

These scripts uses:
- pretrained detection model for weed detection
- unet or sam for initial mask generation
- post segmentation script for refining mask based on either hsv filtreing, EXG thresholding etc.

### CODE REVIEW

**Comment by mkutu** [comment]
* [x] implement `project` structure for `mask_gen` outputs
* [x] implement testing for `mask_gen` pipeline
    - uploading 2-3 smaller images and unet masks to github placing them in a test folder that represent 1-2 mask issues (missing red, missing white)
    - do not save to lts during testing
* [x] update main config to include:
    1. mode and tasks
    2. save2lts flag for mask_inspect
    3. test.enabled
* [x] separate mask issue refinement methods in mask_refine.py into separate refinement scripts

**Comment by mkutu** [comment]
refer to #45 for updates

File: mask_generation/environment.yaml
Comment (mkutu) [inline mask_generation/environment.yaml:1]: remove "version" and add specific package version

**Review by mkutu — APPROVED** [review]
